File: LeetCode/May challenge/Day 23/Day_23.cpp
class Solution {
public:
    vector<vector<int>> intervalIntersection(vector<vector<int>>& A, vector<vector<int>>& B) 
    {
        vector<vector<int>> intersections;
        int i=0, j=0;
        while( i < A.size() && j < B.size() )
        {

            if( ( A[i][0] >= B[j][0] && A[i][0] <= B[j][1] ) ||
                ( B[j][0] >= A[i][0] && B[j][0] <= A[i][1] ) ) {

                intersections.push_back( { max( A[i][0], B[j][0] ), min( A[i][1], B[j][1] ) } );
            }
            if( A[i][1] < B[j][1] ) 
                i++;
            else
                j++;
        }
        return intersections;
    }
};

File: LeetCode/May challenge/Day 23/Day_23.py
class Solution:
    def intervalIntersection(self, A: List[List[int]], B: List[List[int]]) -> List[List[int]]:
        output_intersections = []
        ptr_1 = ptr_2 = 0
        
        while ptr_1 < len(A) and ptr_2 < len(B):
            start_interval = max(A[ptr_1][0], B[ptr_2][0])
            end_interval = min(A[ptr_1][1], B[ptr_2][1])
            
            if start_interval <= end_interval:
                output_intersections.append([start_interval, end_interval])
                
            if A[ptr_1][1] < B[ptr_2][1]:
                ptr_1 += 1
            else:
                ptr_2 += 1
        return output_intersections
